atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -209,26 +214,74 @@ protected boolean shouldStopRetrying(int numTimesFailed) {
         }
     }
 
-    @Override
-    public <T, E extends Exception> T finishRunTaskWithLockThrowOnConflict(TransactionAndImmutableTsLock txAndLock,
-                                                                           TransactionTask<T, E> task)
-            throws E, TransactionFailedRetriableException {
-        Timer postTaskTimer = getTimer("finishTask");
-        Timer.Context postTaskContext;
+    static final OpenTransactions EMPTY = new OpenTransactions() {
+        @Override
+        public List<OpenTransaction> getTransactions() {
+            return ImmutableList.of();
+        }
 
-        TransactionTask<T, E> wrappedTask = wrapTaskIfNecessary(task, txAndLock.immutableTsLock());
+        //        @Override
+        //        public TransactionsLockWatchEvents getEvents(Optional<IdentifiedVersion> lastKnownVersion) {
+        //            throw new SafeIllegalArgumentException("Empty batch");
+        //        }
+    };
 
-        Transaction tx = txAndLock.transaction();
-        T result;
-        try {
-            result = runTaskThrowOnConflict(wrappedTask, tx);
-        } finally {
-            postTaskContext = postTaskTimer.time();
-            timelockService.tryUnlock(ImmutableSet.of(txAndLock.immutableTsLock()));
+    private final class DefaultOpenTransactions implements OpenTransactions {
+
+        private final List<OpenTransaction> transactions;
+
+        DefaultOpenTransactions(
+                List<OpenTransaction> transactions) {
+            this.transactions = transactions;
+        }
+
+
+        @Override
+        public List<OpenTransaction> getTransactions() {
+            return transactions;
+        }
+
+        //    @Override
+        //    public TransactionsLockWatchEvents getEvents(Optional<IdentifiedVersion> lastKnownVersion) {
+        //        return null;
```

Comment:
This would have to work with an empty list of started transactions, and just send the update up to the latest we know about.